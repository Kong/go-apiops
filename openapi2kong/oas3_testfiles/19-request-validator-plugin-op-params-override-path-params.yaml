# When the request-validator is added with path and operation parameters,
# with both sharing same name and location, operation parameters override the path parameters.

openapi: 3.0.2

info:
  title: Example
  version: 1.0.0

servers:
  - url: http://backend.com/path

x-kong-plugin-request-validator: {}

paths:
 /test/common-param/{common-param}:
      parameters:
        - in: path
          name: common-param
          schema:
            type: integer
          required: true
      get:
        parameters:
          - in: path
            name: common-param
            schema:
              type: integer
            required: true
        responses:
          '200':
            description: OK
        x-kong-plugin-request-validator:
          enabled: true
          config:
            verbose_response: true
