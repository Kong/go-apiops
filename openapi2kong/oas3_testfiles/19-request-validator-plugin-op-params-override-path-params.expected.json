{
  "_format_version": "3.0",
  "services": [
    {
      "host": "backend.com",
      "id": "730d612d-914b-5fe8-8ead-e6aa654318ef",
      "name": "example",
      "path": "/path",
      "plugins": [],
      "port": 80,
      "protocol": "http",
      "routes": [
        {
          "id": "bee0be08-646a-562a-91b9-71737169585b",
          "methods": [
            "GET"
          ],
          "name": "example_test-common-param-common-param_get",
          "paths": [
            "~/test/common-param/(?<common_param>[^#?/]+)$"
          ],
          "plugins": [
            {
              "config": {
                "parameter_schema": [
                  {
                    "explode": false,
                    "in": "path",
                    "name": "common_param",
                    "required": true,
                    "schema": "{\"type\":\"integer\"}",
                    "style": "simple"
                  }
                ],
                "verbose_response": true,
                "version": "draft4"
              },
              "enabled": true,
              "id": "3bcb9a87-847d-5ccf-93dc-b2aa1b32b77d",
              "name": "request-validator",
              "tags": [
                "OAS3_import",
                "OAS3file_19-request-validator-plugin-op-params-override-path-params.yaml"
              ]
            }
          ],
          "regex_priority": 100,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_19-request-validator-plugin-op-params-override-path-params.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_19-request-validator-plugin-op-params-override-path-params.yaml"
      ]
    }
  ],
  "upstreams": []
}